### PR TITLE
M876 usbprtdialog A3IDES-594

### PR DESCRIPTION
**Pull request by RadekVana**

### CODE REVIEW

File: src/gui/screen_printing_serial.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+void screen_printing_serial_init(screen_t *screen);
+void screen_printing_serial_done(screen_t *screen);
+void screen_printing_serial_draw(screen_t *screen);
+int screen_printing_serial_event(screen_t *screen, window_t *window, uint8_t event, void *param);
+
+static const char txt0[] = "Remote printing\nin progress    ";
+static const char txt1[] = "Remote printing\nin progress .  ";
+static const char txt2[] = "Remote printing\nin progress .. ";
+static const char txt3[] = "Remote printing\nin progress ...";
```
Comment (DRracer) [inline src/gui/screen_printing_serial.cpp]: this is brutal ... why don't you append the dots to the text when being printed?

File: src/guiapi/src/screen.c [hunk 2]
```diff
@@ -95,6 +95,21 @@ void screen_close(void) {
     }
 }
 
+int _currnet_in_list(screen_t *const *list, size_t sz) {
```
Comment (DRracer) [inline src/guiapi/src/screen.c]: typo?
currnet -> current?

File: src/common/marlin_client.c [hunk 3]
```diff
@@ -49,6 +49,7 @@ typedef struct _marlin_client_t {
     uint32_t command;            // processed command (G28,G29,M701,M702,M600)
     marlin_host_prompt_t prompt; // current host prompt structure (type and buttons)
     uint8_t reheating;           // reheating in progress
+    dialog_cb_t dialog_cb;
```
Comment (DRracer) [inline src/common/marlin_client.c]: document the meaning and usage of this var

File: src/gui/screen_printing_serial.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/screen_printing_serial.cpp]: I did try to redraw just dots. But there is an bug in textfield.
Redrawing dots erased parts of characters around it.

File: src/guiapi/src/screen.c (hunk 2 above)
Comment (RadekVana) [reply src/guiapi/src/screen.c]: Thanks

File: src/common/marlin_client.c (hunk 3 above)
Comment (RadekVana) [reply src/common/marlin_client.c]: comment added

File: src/guiapi/src/screen.c (hunk 2 above)
Comment (RadekVana) [reply src/guiapi/src/screen.c]: fixed

**Pull request merged**